lib/rspec/matchers/built_in/include.rb

```diff
@@ -46,7 +46,12 @@ def failure_message_when_negated
         # @api private
         # @return [Boolean]
         def diffable?
-          true
+          return true unless actual.is_a?(String) && expected.is_a?(Array)
+
+          lines = actual.split("\n")
+          expected.none? do |str|
+            actual.include?(str) && lines.none? { |line| line == str }
+          end
```

Comment:
:+1: I love how simple this is.

 I think it could be worth extracting this logic into a method that describes the what this logic does -- something like `diff_would_wrongly_highlight_matched_item?` -- and then `diffable?` could be defined as `!diff_would_wrongly_highlight_matched_item?`.

Thoughts?